x-pack/plugins/security_solution/public/detections/components/osquery/osquery_flyout.tsx

```diff
@@ -22,11 +29,44 @@ export interface OsqueryFlyoutProps {
   onClose: () => void;
 }
 
-export const OsqueryFlyout: React.FC<OsqueryFlyoutProps> = ({ agentId, onClose }) => {
+const TimelineComponent = React.memo((props) => {
+  return <EuiButtonEmpty {...props} size="xs" />;
+});
+TimelineComponent.displayName = 'TimelineComponent';
+
+export const OsqueryFlyoutComponent: React.FC<OsqueryFlyoutProps> = ({ agentId, onClose }) => {
   const {
-    services: { osquery },
+    services: { osquery, timelines },
   } = useKibana();
 
+  const { getAddToTimelineButton } = timelines.getHoverActions();
+
+  const handleAddToTimeline = useCallback(
+    (payload: { query: [string, string]; isIcon?: true }) => {
+      const [field, value] = payload.query;
```

Comment:
nit: could change this to `const { query: [field, value], isIcon } = payload` so it's more clear that isIcon is used below

Reply:
👍